source/software/singularity.rst

```diff
@@ -0,0 +1,285 @@
+   since if you fail to do so, Singularity will use directories in
+   your home directory, and you will exceed the quota on that file
+   system.
+
+   Also, images tend to be very large, so store them on in a directory
+   where you have sufficient quota, e.g., ``$VSC_DATA``.
+
+Remote builds have several advantages:
+
+- you only need a web browser to create them, so this approach is
+  platform-independent,
+- they can easily be shared with others.
+
+However, local builds still offer more flexibility, especially when
+some interactive setup is required.
+
+
+.. _Singularity definition files:
+
+Singularity definition files
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Below is an example of a Singularity definition file::
+
+   Bootstrap: docker
+   From: ubuntu:xenial
+   
+   %post
+       apt-get update
+       apt-get install -y grace
+               
+   %runscript
+       /usr/bin/grace
+
+The resulting image will be based on the Ubuntu Xenial Xerus distribution
+(18.04).  Once it is bootstrapped, the command in the ``%post`` section of
+the definition file will be executed.  For this example, the Grace plotting
+package will be installed.
+
+.. note::
+
+   This example is intended to illustrate that very old software that
+   is no longer maintained can successfully be run on modern infrastructure.
+   It is by no means intended to encourage you to start using Grace.
+
+Singularity definition files are very flexible, for more details,
+we refer you to the `Singularity definition file documentation`_.
+
+An important advantage of definition files is that they can easily
+be shared, and improve reproducibility.
+
+
+How can I run a Singularity image?
+----------------------------------
+
+Once you have an image, there are several options to run the container.
+
+#. You can invoke any application that is in the ``$PATH`` of the
+   container, e.g., for the image containing Grace::
+
+   $ singularity  exec  grace.sif  xmgrace
+
+#. In case the definition file specified a ``%runscript`` directive,
+   this can be executed using::
+
+   $ singularity  run  grace.sif
+
+#. The container can be run as a shell::
+
+   $ singularity  shell  grace.sif
+
+By default, your home directory in the container will be mounted
+with the same path as it has on the host.  The current working
+directory in the container is that on the host in which you
+invoked ``singularity``.
+
+.. note::
+
+   Although you can move to a parent directory of the current working
+   directory in the container, you will not see its contents on the host.
+   Only the current working directory on the host was mounted.
+
+Additional host directories can be mounted in the container as well by
+using the ``-B`` option.  Mount points are created dynamically (using
+overlays), so they do not have to exist in the image.  For example,
+to mount the ``$VSC_SCRATCH`` directory, you would use::
+
+   $ singularity  exec  -B $VSC_SCRATCH:/scratch  grace.sif  xmgrace
```

Comment:
We (Brussel) by default pass the `$VSC_SCRATCH` and `$VSC_DATA` to the container.

Reply:
Hm, I've mixed feelings about that.  I agree it is convenient for the user,
on the other hand exposing all directories to containers of potentially
suspicious provenance...

Reply:
the `$HOME` and `$PWD` are already passed and this is not different then when the user downloads and executes some piece of software of the internet.

As long as the rights inside and outside the container are the same, I don't see any 'extra' dangers compared to the current situation